src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TabView/TabViewTests.cs

```diff
@@ -0,0 +1,225 @@
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+using Common;
+using Microsoft.UI.Xaml.Controls;
+using System.Collections.Generic;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Provider;
+using Private.Infrastructure;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+using Symbol = Windows.UI.Xaml.Controls.Symbol;
+
+#if !WINDOWS_UWP
+using SymbolIconSource = Windows.UI.Xaml.Controls.SymbolIconSource;
+#else
+using SymbolIconSource = Microsoft.UI.Xaml.Controls.SymbolIconSource;
+#endif
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+
+	[TestClass]
+	public class TabViewTests
+	{
+
+		[TestMethod]
+		public void VerifyCompactTabWidthVisualStates()
+		{
+			TabView tabView = null;
+			RunOnUIThread.Execute(() =>
+			{
+				tabView = new TabView();
+				TestServices.WindowHelper.WindowContent = tabView;
+
+				tabView.TabItems.Add(CreateTabViewItem("Item 0", Symbol.Add));
+				tabView.TabItems.Add(CreateTabViewItem("Item 1", Symbol.AddFriend));
+				tabView.TabItems.Add(CreateTabViewItem("Item 2"));
+
+				tabView.SelectedIndex = 0;
+				tabView.SelectedItem = tabView.TabItems[0];
+				(tabView.SelectedItem as TabViewItem).IsSelected = true;
+				Verify.AreEqual("Item 0", (tabView.SelectedItem as TabViewItem).Header);
+				//TestServices.WindowHelper.WindowContent.UpdateLayout();
+			});
+			// Waiting for layout
+			TestServices.WindowHelper.WaitForIdle();
+
+			RunOnUIThread.Execute(() =>
+			{
+				// Now set tab width mode
+				tabView.TabWidthMode = TabViewWidthMode.Compact;
+			});
+
+			TestServices.WindowHelper.WaitForIdle();
+
+			// Check if switching to compact updates all items correctly
+			RunOnUIThread.Execute(() =>
+			{
+				VerifyTabWidthVisualStates(tabView.TabItems, true);
+				tabView.TabItems.Add(CreateTabViewItem("Item 3"));
+			});
+
+			TestServices.WindowHelper.WaitForIdle();
+
+			// Check if a newly added item has correct visual states
+			RunOnUIThread.Execute(() =>
+			{
+				VerifyTabWidthVisualStates(tabView.TabItems, true);
+				tabView.TabWidthMode = TabViewWidthMode.Equal;
+			});
+
+			TestServices.WindowHelper.WaitForIdle();
+
+			// Switch back to non compact and check if every item has the correct visual state
+			RunOnUIThread.Execute(() =>
+			{
+				VerifyTabWidthVisualStates(tabView.TabItems, false);
+			});
+		}
+
+		[TestMethod]
+		public void VerifyTabViewUIABehavior()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				TabView tabView = new TabView();
+				TestServices.WindowHelper.WindowContent = tabView;
+
+				tabView.TabItems.Add(CreateTabViewItem("Item 0", Symbol.Add));
+				tabView.TabItems.Add(CreateTabViewItem("Item 1", Symbol.AddFriend));
+				tabView.TabItems.Add(CreateTabViewItem("Item 2"));
+
+				//Content.UpdateLayout();
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)